segmenters/nlp/SpacySentencizer/__init__.py

```diff
@@ -0,0 +1,96 @@
+from typing import Dict, List
+
+from jina.executors.segmenters import BaseSegmenter
+
+
+class SpacySentencizer(BaseSegmenter):
+    """
+    :class:`SpacySentencizer` split the text on the doc-level
+    into sentences on the chunk-level with spaCy as the backend.
+    The splitting technique is divided into two different approaches:
+    1) Default Segmenter: Utilizes dependency parsing to determine the rule for
+        splitting the text. For more info please refer to https://spacy.io/api/sentencizer.
+    2) Machine Learning-based Segmenter: Utilizes SentenceRecognizer model trained
+        on a dedicated data created for sentence segmentation. For more info please
+        refer to https://spacy.io/api/sentencerecognizer.
+
+    :param lang: pre-trained spaCy language pipeline
+    :param use_default_segmenter: boolean to determine which segmentation algorithm to use,
+        by default False.
+    :param args:  Additional positional arguments
+    :param kwargs: Additional keyword arguments
+    """
+
+    SPACY_COMPONENTS = [
+        "tagger",
+        "parser",
+        "ner",
+        "senter",
+        "entity_linker",
+        "textcat",
+        "entity_ruler",
+        "merge_noun_chunks",
+        "merge_entities",
+        "merge_subtokens",
+        "lemmatizer",
+        "attribute_ruler",
+    ]
+
+    def __init__(self, lang: str, use_default_segmenter: bool = False, *args, **kwargs):
+        """Set constructor."""
+        super().__init__(*args, **kwargs)
+        self.lang = lang
+        self.use_default_segmenter = use_default_segmenter
+
+    def post_init(self):
+        import spacy
+
+        try:
+            self.spacy_model = spacy.load(self.lang)
+            # Disable everything as we only requires certain pipelines to turned on.
+            ignored_components = []
+            for comp in self.SPACY_COMPONENTS:
+                try:
+                    self.spacy_model.disable_pipe(comp)
+                except Exception:
+                    ignored_components.append(comp)
+            self.logger.info(f"Ignoring {ignored_components} pipelines as it does not available on the model package.")
```

Comment:
Please use single quote when possible

Reply:
just out of curiosity, why single quote?